dom/data/data-test.js

```diff
@@ -123,3 +123,28 @@ unit.test('domData should be cleaned up if element is removed from DOM after cal
 
 	checkRemoved();
 });
+
+// MutationObserver is needed for this test
+if(typeof MutationObserver !== 'undefined'){
+	unit.test('domData should count active elements', function (assert){
+		var done = assert.async();
+		var div = document.createElement('div');
+		var fixture = document.getElementById('qunit-fixture');
+		var startingCount = domData._elementSetCount;
+	
+		fixture.appendChild(div);
+		domData.set.call(div, 'foo', 'bar');
+	
+		assert.equal(domData.get.call(div, 'foo'), 'bar', 'foo was set');
+		assert.equal(domData._elementSetCount, startingCount + 1, '_elementSetCount incremented');
+	
+		fixture.removeChild(div);
+	
+		// Remove event fires on next tick
+		setTimeout(function(){
```

Comment:
Instead of using `setTimeout` I would try to use the `mutationDocument.onAfterRemovedNodes`. Not only does this document the test's dependency, but it also is less finicky than a setTimeout as MO mutations are asynchronous and not always on the very next tick.